### ISSUE TITLE
Naming conventions

### ISSUE THREAD

**Issue by mechanical-turk** [VulcanJS/Vulcan#1699]
# Background
This is an extension of a recent discussion we had over slack. The reason I'm creating the git issue is because it took us some while to realize that the permissions feature accepts all lowercase inputs, although we have never declared anything of the sorts. We believe it's generated automatically, and it might cause a lot of confusion, especially for beginners. We were doing the movies example, but instead of `movies`, we did `todosLists`. 

Here's how the collection looked:
```js
const TodosLists = createCollection({
  schema,
  collectionName: 'TodosLists',
  typeName: 'TodosList',
  resolvers: getDefaultResolvers('TodosLists'),
  mutations: getDefaultMutations('TodosLists'),
});
```

And here's how our permissions looked:
```js
Users.groups.members.can([
  'TodosLists.new',
  'TodosLists.own.edit',
]);
```

However, it did not work correctly, until we tried a few combinations and came up with the following:

```js
Users.groups.members.can([
  'todoslists.new',
  'todoslists.own.edit',
]);
```

If you check out your mongodb collections by running `meteor mongo`, you'll also see the same "all lowercase" phenomenon going. This all points to a confusing pattern where things get named differently behind the scenes, and it clashes with the smooth dev experience that Vulcan is meant to bring. So I propose that we create our naming conventions, once and for all, and just stick to them.

Once we come to a conclusion, I will hardcode the naming conventions into the `CLI`.

# Logical Reasoning

First of all, we need to be on the same page in terms of **how** things are named. For example, we thought that `permissions` should be named after the `collectionName`. But it was't. There was some other background logic that we weren't aware of. So my first suggestion is to establish **who defines what**. I'm gonna write what makes sense to me, in the following form: `A depends on B`. For example: `resolvers` depends on `collectionName`. This way, we can establish a `naming dependency tree` of some sorts, and just pick a name for the root of the tree, and then just follow the conventions to figure out the rest of the naming. So here's what I believe how the dependency tree should look like:

- `typeName` is the root
- `collectionName` depends on `typeName`
- `resolvers` depends on `collectionName`
- `mutations` depends on `collectionName`
- `parameters` depends on `collectionName`
- `permissions` depends on `collectionName`
- `mongodb collection name` depends on `collectionName`

# Naming conventions

My intention is to open the discussion on 2 dimensions: Casing and Plurality. 
- How should the names be cased: alllowercase, camelCase, dashed-lowercase, underscore_lowercase, PascalCase
- Whether names should be plural or singular

I think that the simpler we go, the better. For those reasons, I suggest that everything be `PascalCased`, and almost everything be `plural`.

- `typeName`: `PascalCase`, `Singular`
- `collectionName`: `PascalCase`, `Plural`

And since everything else depends on the `collectionName`, they by extension should get `PascalCase` + `Plural` names.

# Naming examples
Combining everything above, we get the following conclusion: Pick a `typeName`, and the rest will take care of itself automatically. So let's do a todo list example. Let's say that we can to create a todo list model.

- `typeName`: (`root`) (`PascalCase` + `Singular`) TodoList
- `collectionName`: (plural of `typeName`)(`PascalCase` + `Plural`) TodoLists
- `resolvers`: (`collectionName` + `PascalCase resolver type`)(`PascalCase` + `Plural`) TodoListsTotal, TodoListsSingle, TodoListsList
- `mutations`: (`collectionName` + `.` + `camelCase action name`)(`PascalCase` + `Plural` + `camelCase`) TodoLists.new, TodoLists.edit, TodoLists.remove
- `parameters`: (Identical to `collectionName`): TodoLists
- `permissions`: (Identical to `collectionName`): TodoLists
- `mongodb collection name`: (Identical to `collectionName`): TodoLists

# Conclusion
I realize that creating a naming convention at this stage may break old projects, but we will eventually have to establish them. I believe this will create a lot of technical debt, and the sooner we figure it out, the better.

I also realize that the naming suggestions above may look a little arbitrary. And they are actually arbitrary. Most naming conventions are. However, as long as they are simple, and dont have the devs look up some table all the time, I say they are getting the job done.

Sorry for the long post. I hope that this will start a discussion, and we can finalize this issue.

**Comment by SachaG** [comment]
Yeah, that makes sense to me. We could also just make `Users.canDo` convert everything to lowercase itself, so that both `TodoLists.edit` and `todolists.edit` would work? Seems like a good idea, at least from a backwards-compatibility point of view. 

Btw the MongoDB collection name will always be lowercase, that's just how Meteor (or maybe Mongo itself?) does it.

**Comment by mechanical-turk** [comment]
yeah i remembered the mongo thing now. converting everything to a standard form in the background seems like a good idea
